crates/sqlbuiltins/src/functions/scalars/sentence_transformers.rs

```diff
@@ -0,0 +1,253 @@
+};
+use datafusion::physical_plan::ColumnarValue;
+use datafusion::scalar::ScalarValue;
+use hf_hub::api::sync::Api;
+use hf_hub::{Repo, RepoType};
+use protogen::metastore::types::catalog::FunctionType;
+use tokenizers::{PaddingParams, Tokenizer};
+
+use crate::functions::{BuiltinScalarUDF, ConstBuiltinFunction, FunctionNamespace};
+
+#[derive(Clone)]
+pub struct BertUdf {
+    model: Arc<BertModel>,
+    signature: Signature,
+    tokenizer: Tokenizer,
+    device: Device,
+    output_dim: usize,
+}
+
+impl std::fmt::Debug for BertUdf {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BertUdf").finish()
+    }
+}
+
+impl BertUdf {
+    pub fn new() -> Self {
+        let device = Device::Cpu;
+        let model_id = "sentence-transformers/all-MiniLM-L6-v2".to_string();
+        let revision = "refs/pr/21".to_string();
+
+        let repo = Repo::with_revision(model_id, RepoType::Model, revision);
+
+        let (config_filename, tokenizer_filename, weights_filename) = {
+            let api = Api::new().expect("failed to create api");
+            let api = api.repo(repo);
+            let config = api.get("config.json").expect("failed to get config");
+            let tokenizer = api.get("tokenizer.json").expect("failed to get tokenizer");
+            let weights = api.get("model.safetensors").expect("failed to get weights");
+
+            (config, tokenizer, weights)
+        };
+        let config = std::fs::read_to_string(config_filename).expect("failed to read config");
+
+
+        let config: Config = serde_json::from_str(&config).expect("failed to parse config");
+        let mut tokenizer =
+            Tokenizer::from_file(tokenizer_filename).expect("failed to load tokenizer");
+
+        let pp = PaddingParams {
+            strategy: tokenizers::PaddingStrategy::BatchLongest,
+            ..Default::default()
+        };
+        tokenizer.with_padding(Some(pp));
+        let vb = unsafe {
+            VarBuilder::from_mmaped_safetensors(&[weights_filename], DTYPE, &device)
+                .expect("failed to load weights")
+        };
+        let model = BertModel::load(vb, &config).expect("failed to load model");
+        let signature = Signature::exact(vec![DataType::Utf8], Volatility::Stable);
+
+        let output_dim = 384;
+
+        Self {
+            model: Arc::new(model),
+            signature,
+            tokenizer,
+            device,
+            output_dim,
+        }
+    }
+}
+
+impl ConstBuiltinFunction for BertUdf {
+    const NAME: &'static str = "minilm_l6_v2";
+
+    const DESCRIPTION: &'static str = "Transforms a sentence into a vector using a BERT model";
+
+    const EXAMPLE: &'static str = "sentence_transformers.minilm_l6_v2('This is a sentence')";
+
+    const FUNCTION_TYPE: FunctionType = FunctionType::Scalar;
+
+    fn signature(&self) -> Option<Signature> {
+        Some(self.signature.clone())
+    }
+}
+
+impl ScalarUDFImpl for BertUdf {
```

Comment:
[curiosity] what's the reason for chosing to implement this trait rather than BuiltinUDF (as the other scalar functions are implemented?)

Reply:
No sense in adding more tech debt. We're already going to have to migrate to `ScalarUDFImpl` at some point anyways.

Reply:
Having different ways of implementing these functions is also adding technical debt if the transition plan is long.